src/EFCore.PG.NodaTime/Query/ExpressionTranslators/Internal/NpgsqlNodaTimeMemberTranslatorPlugin.cs

```diff
@@ -126,17 +139,21 @@ static Expression TranslateDateTime([NotNull] MemberExpression e)
                 // Unlike DateTime.DayOfWeek, NodaTime's IsoDayOfWeek enum doesn't exactly correspond to PostgreSQL's
                 // values returned by DATE_PART('dow', ...): in NodaTime Sunday is 7 and not 0, which is None.
                 // So we generate a CASE WHEN expression to translate PostgreSQL's 0 to 7.
-                var getValueExpression = GetDatePartExpression(e, "dow", true);
+                var getValueExpression = GetDatePartExpression(instance, "dow", true);
+                // TODO: Can be simplified once https://github.com/aspnet/EntityFrameworkCore/pull/16726 is in
                 return
-                    Expression.Condition(
-                        Expression.Equal(
-                            getValueExpression,
-                            Expression.Constant(0)),
-                        Expression.Constant(7),
-                        getValueExpression);
+                    _sqlExpressionFactory.Case(
+                        new[]
+                        {
+                            new CaseWhenClause(
+                                _sqlExpressionFactory.Equal(getValueExpression, _sqlExpressionFactory.Constant(0)),
+                                _sqlExpressionFactory.Constant(7))
+                        },
+                        getValueExpression
+                    );
 
             case "Date":
-                return new SqlFunctionExpression("DATE_TRUNC", e.Type, new[] { Expression.Constant("day"), e.Expression });
+                return _sqlExpressionFactory.Function("DATE_TRUNC", new[] { _sqlExpressionFactory.Constant("day"), instance }, returnType);
```

Comment:
DATE_TRUNC accepts an actual (quoted) string - it can even be an expression (i.e. can send a string parameter). Am trying to avoid fragments as much as possible as discussed :)